pkg/cloudsources/discoveredclusters/discovered_cluster.go

```diff
@@ -0,0 +1,119 @@
+package discoveredclusters
+
+import (
+	"strings"
+
+	gogoTimestamp "github.com/gogo/protobuf/types"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/errorhelpers"
+	"github.com/stackrox/rox/pkg/uuid"
+)
+
+var rootNamespaceUUID = uuid.FromStringOrPanic("1a379e9f-1ea4-4399-b7e1-4cd8b1ced3e3")
+
+// GenerateDiscoveredClusterID returns a UUID5 based on the discovered cluster fields.
+func GenerateDiscoveredClusterID(discoveredCluster *DiscoveredCluster) string {
+	key := strings.Join([]string{
```

Comment:
Perform `nil` check on the argument before processing it.

Reply:
Do you think the nil-safe getters are sufficient here or should we fail explicitly?